text/0000-clarify-api-namespace.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2015-08-10
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+Clarify API description namespace and Iron out issues which rose during API Blueprint serialization to API description namespace.
+
+# Motivation
+The purpose of this document is to clear inconsistencies in the format and make it easier to build tools around it.
+
+# Detailed design
+
+## HREF Variables
+We need to clarify about the value of the `content` in the `hrefVariables` element.
+
+## Data Structure
+We don't need a `Data Structure` because an element which is an `object`, `extend` or `array` can be considered as a data structure element.
+
+*(The current spec is a lot confusing. What is the `Data Structure Element`? Why is `Data Structure` inheriting from it?)*
+
+## Resource
+Since `hrefVariables` is an element, we should move it to be under `content` rather than leave it under `attributes`.
+
+## Transition
+Similarily to the Resource, we need to move `hrefVariables` to be under `content`. `Data Structure` in Resource is under `content`, but it is under `attributes` in Transition. To resolve this inconsistency, we need to move `Data Structure` to be under `content`. Another reason for it is that `Data Structure` is an element.
+
+## Transition Example
+Value of `content` should be changed to `[]` instead of `null`.
+
+## Category
+We need to clarify what `transitions` category is about. If it is being reserved for future use, we need to add reserved keyword.
+
+## Copy
+We need to clarify what the relation between copy elements in `content` and `description` in `attributes` is. Is `description` just an aggregator of all `copy` elements? Why does some examples contain `description` without any `copy` elements and some examples contain `copy` elements but doesn't contain `description`. Since `copy` element is superior, why do we need `description` if we can use `copy` elements everywhere?
```

Comment:
I suggest following: 

> # Copy (Element)

> Copy element represents a copy text – a textual information content of an element. Copy element MAY include information about the media type of the copy's content. 

> Unless specified otherwise, a copy element's content represents the description of it's parent element and SHOULD be used instead of parent element's description metadata.

In addition to this I suggest to explicitly add Copy as a possible value for `content` arrays of:

- Category element
- Resource element
- Transition element
- HTTP Transaction element
- HTTP Message Payload element

---

However, if you are following, it is clear this does not solve MSON – Data Structure Namespace and HTTP headers. The reason for this is that currently it is not possible to add Copy elements to primitive types. 

Consider theoretical future syntax (`description` keyword): 

```apib
# Person (object)
## Description
Copy1 

## Properties
- name

## Description
Copy 2
```

With current refract namespace this will cause us a lot of headache. Not necessary a problem of the `copy` element discussed here, but something we need to consider.